src/nvim/os/keyboard.c

```diff
@@ -82,12 +165,48 @@ bool capslock_status(const char **const err )
   IOObjectRelease(ios);
 
   if (IOHIDGetModifierLockState(ioc, kIOHIDCapsLockState,
-                                &state) != kIOReturnSuccess) {
+                                &stateCaps) != kIOReturnSuccess) {
     *err = "Unable to lock the state";
     return false;
   }
-  IOServiceClose(ioc);
-  return (bool)state;
 
+  if (IOHIDGetModifierLockState(ioc, kIOHIDCapsLockState,
+                                &stateNums) != kIOReturnSuccess) {
+    *err = "Unable to lock the state";
+    return false;
+  }
+  if (stateCaps) {
+    *mods = *mods | kCapsLock;
+  }
+  if (stateNums) {
+    *mods = *mods | kNumLock;
+  }
+  IOServiceClose(ioc);
+  return true;
+}
 #endif
+
+/// A helper function to be used by `nvim_get_caps_state` API function giving
+/// the information about the Capslock state.
+///
+/// @param err[out]  The error if encountered, otherwise will be NULL
+/// @return          The status of capslock, `true` if active otherwise `false`
+bool mods_status(Dictionary *const dict, const char **const err )
```

Comment:
All functions in `os/*` should be prefixed with `os_`.

Reply:
Ok I will do it, now traveling, will do in hour or so